katharsis-client/src/main/java/io/katharsis/client/internal/RequestUrlBuilder.java

```diff
@@ -0,0 +1,95 @@
+package io.katharsis.client.internal;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Map;
+
+import com.squareup.okhttp.HttpUrl;
+import com.squareup.okhttp.HttpUrl.Builder;
+
+import io.katharsis.queryParams.QueryParams;
+import io.katharsis.resource.information.ResourceInformation;
+import io.katharsis.resource.registry.RegistryEntry;
+import io.katharsis.resource.registry.ResourceRegistry;
+import io.katharsis.utils.StringUtils;
+
+public class RequestUrlBuilder {
+
+	private QueryParamsSerializer queryParamsSerializer = new DefaultQueryParamsSerializer();
+
+	private ResourceRegistry resourceRegistry;
+
+	public RequestUrlBuilder(ResourceRegistry resourceRegistry) {
+		this.resourceRegistry = resourceRegistry;
+	}
+
+	public <T> HttpUrl buildUrl(Class<T> resourceClass, Object id, QueryParams queryParams) {
+		return buildUrl(resourceClass, id, queryParams, null);
+	}
+	
+	public <T> HttpUrl buildUrl(Class<T> resourceClass, Object id, QueryParams queryParams, String relationshipName) {
+		String strRepoUrl = resourceRegistry.getResourceUrl(resourceClass);
+		if (!strRepoUrl.endsWith("/")) {
+			strRepoUrl += "/";
+		}
+		HttpUrl repositoryUrl = HttpUrl.parse(strRepoUrl);
+
+		HttpUrl.Builder urlBuilder = new HttpUrl.Builder();
+		urlBuilder.host(repositoryUrl.host());
+		urlBuilder.port(repositoryUrl.port());
+		urlBuilder.scheme(repositoryUrl.scheme());
+		
+		RegistryEntry<?> entry = resourceRegistry.getEntry(resourceClass);
+		ResourceInformation resourceInformation = entry.getResourceInformation();
+		
+		String encodedPath;
+		if (id instanceof Collection) {
+			Collection<?> ids = (Collection<?>) id;
+			Collection<String> strIds = new ArrayList<String>();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Replace the type specification in this constructor call with the diamond operator ("<>"). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS2293)